plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/hadoop/TrinoHadoopCatalog.java

```diff
@@ -299,31 +338,81 @@ public void unregisterTable(ConnectorSession session, SchemaTableName tableName)
     @Override
     public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
-        getCatalog(session).dropTable(toTableId(schemaTableName), true);
+        TableOperations ops = newTableOps(session, Optional.empty(), schemaTableName);
+        try {
+            // Since the data files and the metadata files may store in different locations,
+            // so it has to call dropTableData to force delete the data file.
+            CatalogUtil.dropTableData(ops.io(), ops.current());
+            trinoFileSystem.deleteDirectory(Location.of(tableLocation(schemaTableName)));
+        }
+        catch (IOException e) {
+            throw new TrinoException(GENERIC_INTERNAL_ERROR, "Failed to delete table: " + schemaTableName, e);
+        }
+//        invalidateTableCache(schemaTableName);
     }
 
     @Override
     public void dropCorruptedTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
-        getCatalog(session).dropTable(toTableId(schemaTableName), true);
+        dropTable(session, schemaTableName);
     }
 
     @Override
     public void renameTable(ConnectorSession session, SchemaTableName from, SchemaTableName to)
     {
-        getCatalog(session).renameTable(toTableId(from), toTableId(to));
-    }
+        throw new TrinoException(NOT_SUPPORTED, "renameTable is not supported by " + catalogName);
+    }
+
+//    @Override
+//    public Table loadTable(ConnectorSession session, SchemaTableName schemaTableName)
+//    {
+//        TableMetadata metadata;
+//        try {
+//            metadata = uncheckedCacheGet(
+//                    tableMetadataCache,
+//                    schemaTableName,
+//                    () -> {
+//                        IcebergTableOperations operations = tableOperationsProvider.createTableOperations(
+//                                this,
+//                                session,
+//                                schemaTableName.getSchemaName(),
+//                                schemaTableName.getTableName(),
+//                                Optional.empty(),
+//                                Optional.ofNullable(tableLocation(schemaTableName)));
+//                        operations.refresh();
+//                        return new BaseTable(operations, quotedTableName(schemaTableName), TRINO_METRICS_REPORTER).operations().current();
+//                    });
+//        }
+//        catch (UncheckedExecutionException e) {
+//            throwIfUnchecked(e.getCause());
+//            throw new NoSuchTableException("Table does not exist at location: %s", tableLocation(schemaTableName));
+//        }
+//
+//        return getIcebergTableWithMetadata(this, tableOperationsProvider, session, schemaTableName, metadata);
+//    }
```

Comment:
This whole block should be removed

Reply:
This one I want to keep as I hope we will implement the caching and I am pretty sure this code will work without major modifications. Other additions will be required to make table metadata caching work but I strongly believe this is a good start and it should be implemented. Should not require too much effort either I don't think.